tests/sentry/sentry_apps/token_exchange/test_grant_exchanger.py

```diff
@@ -35,13 +35,8 @@ def test_adds_token_to_installation(self):
         token = self.grant_exchanger.run()
         assert SentryAppInstallation.objects.get(id=self.install.id).api_token == token
 
-    @patch("sentry.mediators.token_exchange.Validator.run")
-    def test_validate_generic_token_exchange_requirements(self, validator):
-        self.grant_exchanger.run()
-
-        validator.assert_called_once_with(
-            install=self.install, client_id=self.client_id, user=self.user
-        )
+    def test_validate_generic_token_exchange_requirements(self):
+        assert self.grant_exchanger._validate()
```

Comment:
Having tests call 'protected methods' can make tests brittle as code changes over time.  The mock solution also had this weakness though. It looks like we have integration tests for the validation flow below though, and we could just delete these tests without losing any coverage.